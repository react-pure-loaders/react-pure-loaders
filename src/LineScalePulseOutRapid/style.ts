import { keyframes } from '@emotion/react';
import { globalAnimation, lines } from '../mixins';

const lineScalePulseOutRapid = keyframes({
  '0%': {
    transform: 'scaley(1.0)',
  },
  '80%': {
    transform: 'scaley(0.3)',
  },
  '90%': {
    transform: 'scaley(1.0)',
  },
});

export default (color?: string) => ({
  ' > div': {
    ...lines(color),
    ...globalAnimation(),
    animation: `${lineScalePulseOutRapid} 0.9s -0.5s infinite cubic-bezier(.11,.49,.38,.78)`,
    display: 'inline-block',
    verticalAlign: 'middle',

    '&:nth-of-type(2), &:nth-of-type(4)': {
      animationDelay: '-0.25s !important',
    },

    '&:nth-of-type(1), &:nth-of-type(5)': {
      animationDelay: '0s !important',
    },
  },
});
