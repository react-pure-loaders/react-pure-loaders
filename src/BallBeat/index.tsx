/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
