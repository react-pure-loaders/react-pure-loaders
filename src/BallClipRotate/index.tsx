/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
