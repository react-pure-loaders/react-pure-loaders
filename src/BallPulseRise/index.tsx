/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
