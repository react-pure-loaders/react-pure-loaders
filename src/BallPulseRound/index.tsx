/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
