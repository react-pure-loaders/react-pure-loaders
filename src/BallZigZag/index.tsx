/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
