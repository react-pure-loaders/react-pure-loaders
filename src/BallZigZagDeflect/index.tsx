/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
