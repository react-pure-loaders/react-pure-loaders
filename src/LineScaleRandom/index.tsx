/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
