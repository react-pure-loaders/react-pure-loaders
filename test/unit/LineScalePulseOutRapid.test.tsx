import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import LineScalePulseOutRapid from '../../src/LineScalePulseOutRapid';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<LineScalePulseOutRapid>', () => {
    afterEach(cleanup);

    test('LineScalePulseOutRapid should match snapshot', () => {
        const { container } = render(<LineScalePulseOutRapid loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('LineScalePulseOutRapid should have default color', () => {
        const { container } = render(<LineScalePulseOutRapid loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test.skip('LineScalePulseOutRapid should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<LineScalePulseOutRapid color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('LineScalePulseOutRapid should have no children', () => {
        const { container } = render(<LineScalePulseOutRapid loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('LineScalePulseOutRapid should have three children', () => {
        const { container } = render(<LineScalePulseOutRapid loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(6);
    });
});
