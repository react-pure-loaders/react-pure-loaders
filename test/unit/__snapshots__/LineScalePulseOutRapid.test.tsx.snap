// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineScalePulseOutRapid> LineScalePulseOutRapid should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  80% {
    -webkit-transform: scaley(0.3);
    -moz-transform: scaley(0.3);
    -ms-transform: scaley(0.3);
    transform: scaley(0.3);
  }

  90% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }
}

.emotion-0>div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 0.9s -0.5s infinite cubic-bezier(.11,.49,.38,.78);
  animation: animation-0 0.9s -0.5s infinite cubic-bezier(.11,.49,.38,.78);
  display: inline-block;
  vertical-align: middle;
}

.emotion-0>div:nth-of-type(2),
.emotion-0>div:nth-of-type(4) {
  -webkit-animation-delay: -0.25s!important;
  animation-delay: -0.25s!important;
}

.emotion-0>div:nth-of-type(1),
.emotion-0>div:nth-of-type(5) {
  -webkit-animation-delay: 0s!important;
  animation-delay: 0s!important;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
